source/logic/cooldowns.js

```diff
@@ -0,0 +1,57 @@
+const { Sequelize, Op } = require("sequelize");
+const { dateInPast } = require("../shared");
+const { GLOBAL_COMMAND_COOLDOWN } = require("../constants");
+
+/** @type {Sequelize} */
+let db;
+
+function setDB(database) {
+	db = database;
+}
+
+async function checkCooldownState(userId, interactionName, interactionTime) {
+	const allInteractions = await db.models.UserInteraction.findOne({ where: { userId }, order: [[ "cooldownTime", "DESC" ]] });
+	if (allInteractions && allInteractions.interactionTime + GLOBAL_COMMAND_COOLDOWN < interactionTime) {
+		return {
+			isOnGeneralCooldown: true,
+			isOnCommandCooldown: false,
+			cooldownTimestamp: allInteractions.interactionTime + GCD,
+			lastCommandName: allInteractions.interactionName
+		};
+	}
+	const thisInteractions = await db.models.UserInteraction.findOne({ where: { userId, interactionName } , order: [[ "cooldownTime", "DESC" ]]});
+	if (thisInteractions && thisInteractions.cooldownTime < interactionTime) {
+		return {
+			isOnGeneralCooldown: false,
+			isOnCommandCooldown: true,
+			cooldownTimestamp: thisInteractions.cooldownTime,
+			lastCommandName: interactionName
+		};
+	}
+}
+
+async function updateCooldowns(userId, interactionName, interactionTime, interactionCooldown) {
+	const [interaction, wasCreated] = await db.models.UserInteraction.findOrCreate({ where: { userId, interactionName } });
```

Comment:
```suggestion
	const [interaction, wasCreated] = await db.models.UserInteraction.findOrCreate({ where: { userId, interactionName }, defaults: { hitTimes: 1 } });
```
Adding this defaults option will make it so this `UserInteraction` is created with a `hitTimes` value of 1. Alternatively, I think we could also make the `defaultValue` of `hitTimes` 1 in the schema, since I can't think of a time we'd want to initialize a row with a `hitTimes` value of 0.

Reply:
Alternatively alternatively, unconditionally incrementing `hitTimes` trades a database ping for a logic simplification.

Reply:
The idea was to default to 0 so that it is easier to do future math regarding added cooldown time with additional hits (multiplying additional cooldown time by number of hits on that command).